client/src/app/site/motions/components/category-list/category-list.component.ts

```diff
@@ -147,18 +144,17 @@ export class CategoryListComponent extends BaseComponent implements OnInit {
     /**
      * Saves the categories
      */
-    public onSaveButton(viewCategory: ViewCategory): void {
+    public async onSaveButton(viewCategory: ViewCategory): Promise<void> {
```

Comment:
Doesn't this cause issues with a "long line" to the server with the js not responding? Are Promises in the Front-End area of the code really a good-practice?

Reply:
There shouldn't be any Promises returned in the Repositories. Observables are needed by the Front-End Functions and in the Front-End shouldn't be any promises. Also: Auto-Updates are likely to not work on a Promise-Based systems, because you don't receive automatic updates (like in observables) on changes of the data-set.